openid4vc-high-assurance-interoperability-profile-1_0.md

```diff
@@ -217,7 +217,10 @@ The following requirements apply to OpenID for Verifiable Presentations via redi
 
 * As a way to invoke the Wallet, the custom URL scheme `haip-vp://` MAY be supported by the Wallet and the Verifier. Implementations MAY support other ways to invoke the Wallets as agreed upon by trust frameworks/ecosystems/jurisdictions, including but not limited to using other custom URL schemes or claimed "https" scheme URIs.
 * Signed Authorization Requests MUST be used by utilizing JWT-Secured Authorization Request (JAR) [@!RFC9101] with the `request_uri` parameter.
-* Response encryption MUST be used by utilizing response mode `direct_post.jwt`, as defined in Section 8.3 of [@!OIDF.OID4VP]. Security Considerations in Section 14.3 of [@!OIDF.OID4VP] MUST be applied.
+* Response encryption MUST be used by utilizing response mode `direct_post.jwt`, as defined in Section 8.3 of [@!OIDF.OID4VP]. Security considerations in Section 14.3 of [@!OIDF.OID4VP] MUST be applied.
+* The Verifier MUST reject flows that are non "Same-Device". It MUST do this by including `redirect_uri` in the HTTP response to the Wallet's HTTP POST to the `response_uri`, as defined in Section 8.2 of [@!OIDF.OID4VP]. Wallets MUST follow the the redirect to `redirect_uri`. Verifiers MUST reject presentations if Wallets do not follow the redirect back or the redirect back arrives in a different user session to the one the request was initiated in. Implementation considerations can be found in Section 13.3 of [@!OIDF.OID4VP] and security considerations in Section 14.2 of [@!OIDF.OID4VP].
+
+Note: This means cross-device flows using redirect based [@!OIDF.OID4VP] are not permitted in this profile. Where available, it is recommended to use (#oid4vp-dc-api) which allows secure cross device flows. Where the Digital Credentials API (or a platform equivalent) is not available, Verifiers can achieve a cross-device flow by securely transferring the user session to the other device before initiating the OID4VP flow. In this case, Verifiers MUST establish that the same user is interacting with both devices, for example by requiring the user to perform authentication on both devices and verifying the same account is used in both sessions.
```

Comment:
```suggestion
Note: This means cross-device flows using redirect based [@!OIDF.OID4VP] are not permitted in this profile. Where available, it is recommended to use (#oid4vp-dc-api) which allows secure cross device flows. Where the Digital Credentials API (or a platform equivalent) is not available, Verifiers can achieve a cross-device flow by securely transferring the user session to the other device before initiating the OpenID4VP flow. In this case, Verifiers MUST establish that the same user is interacting with both devices, for example by requiring the user to perform authentication on both devices and verifying the same account is used in both sessions.
```

Reply:
@paulbastian I guess you intended to commit this change but you haven't actually done so?